app/javascript/react/screens/App/Plan/components/PlanRequestDetailList/PlanRequestDetailList.js

```diff
@@ -270,6 +277,70 @@ class PlanRequestDetailList extends React.Component {
     }
   };
 
+  getCancelSelectionState = () => {
+    const { selectedCancelTasks } = this.state;
+    const filteredTasks = this.filterPlanRequestTasks();
+    const incompleteTasks = filteredTasks.filter(task => !task.completed && !task.cancel_requested);
+    return {
+      filteredTasks,
+      incompleteTasks,
+      allSelected: selectedCancelTasks.length === incompleteTasks.length,
+      noneSelected: selectedCancelTasks.length === 0
+    };
+  };
+
+  handleCheckboxChange = task => {
+    const { selectedCancelTasks } = this.state;
+    const selectedTask = selectedCancelTasks.find(t => t.id === task.id);
+    let updatedSelectedTasks;
+    if (selectedTask) {
+      updatedSelectedTasks = selectedCancelTasks.filter(r => !(r.id === task.id));
+    } else {
+      updatedSelectedTasks = [...selectedCancelTasks, task];
+    }
+    this.setState({ selectedCancelTasks: updatedSelectedTasks });
+  };
+
+  handleSelectAllCheckboxChange = () => {
+    const { allSelected } = this.getCancelSelectionState();
+    if (allSelected) {
+      this.deselectAllTasks();
+    } else {
+      this.selectAllInProgressTasks();
+    }
+  };
+
+  taskIsSelected = task => {
+    const { selectedCancelTasks } = this.state;
+    return selectedCancelTasks.findIndex(t => t.id === task.id) > -1;
+  };
+
+  selectAllInProgressTasks = () => {
+    const { incompleteTasks } = this.getCancelSelectionState();
+    this.setState({ selectedCancelTasks: incompleteTasks });
+  };
+
+  deselectAllTasks = () => {
+    this.setState({ selectedCancelTasks: [] });
+  };
+
+  onCancelMigrationsCancel = () => {
+    this.setState({ showConfirmCancel: false });
+  };
+
+  onCancelMigrationsClick = () => {
+    this.setState({ showConfirmCancel: true });
+  };
+
+  onCancelMigrationsConfirm = () => {
+    // todo: call this from the Confirmation Modal instead
+    // gather the selected tasks in state and feed them through
+    const { selectedCancelTasks } = this.state;
+    const { cancelPlanRequestTasksAction, cancelPlanRequestTasksUrl } = this.props;
+    cancelPlanRequestTasksAction(cancelPlanRequestTasksUrl, selectedCancelTasks);
```

Comment:
We might want to refetch the tasks here.  Currently, after confirming the cancel, the success notification appears, but we need to wait for the next poll before the ListItem is updated

Reply:
did not do a refetch here.

The listview reflects the cancelled state immediately upon a successful return from the API POST `cancel` action.
I have used `markedForCancellation` redux state to mark the cancelled items in the listview before the task is officially marked with `cancel_requested=true` by the API.